pkg/db/db.go

```diff
@@ -0,0 +1,117 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+)
+
+func getConnectionString(db *config.DB) string {
+	return fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", db.Host, db.Port, db.User, db.Password, db.Name)
+}
+
+type Database struct {
+	tableName string
+	pool      *pgxpool.Pool
+}
+
+func NewClient(ctx context.Context, db *config.DB) (*Database, error) {
+	connString := getConnectionString(db)
+	pool, err := pgxpool.New(ctx, connString)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := pool.Ping(ctx); err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		pool:      pool,
+		tableName: db.ProvidersTableName,
+	}, nil
+}
+
+func (pg *Database) Close() {
+	pg.pool.Close()
+}
+
+var once sync.Once
+
+func (pg *Database) ListenForNotifications(ctx context.Context, db *config.DB, reqChan chan types.Request) error {
+	var pgConn *pgconn.PgConn
+	var err error
+	once.Do(func() {
+		connString := getConnectionString(db)
+		var config *pgconn.Config
+		if config, err = pgconn.ParseConfig(connString); err != nil {
+			return
+		}
+		config.OnNotification = func(c *pgconn.PgConn, n *pgconn.Notification) {
+			req := types.Request{}
+			if err := json.Unmarshal([]byte(n.Payload), &req); err != nil {
+				return
+			}
+			reqChan <- req
+		}
+
+		if pgConn, err = pgconn.ConnectConfig(ctx, config); err != nil {
+			return
+		}
+		query := fmt.Sprintf("LISTEN %s;", db.ProvidersTableName)
+		// Listen for notifications
+		if _, err = pgConn.Exec(ctx, query).ReadAll(); err != nil {
+			pgConn.Close(ctx)
+			return
+		}
+	})
+	if err != nil {
+		return err
+	}
+
+	if err := pgConn.WaitForNotification(ctx); err != nil {
+		pgConn.Close(ctx)
+		return err
+	}
```

Comment:
@Madhu-1 as you stated in one of your prev comments. this need to run in an infinite loop